_posts/2020-08-05-hive-acid-original-files.md

```diff
@@ -0,0 +1,130 @@
+files and then goes into details of  read support for such files that was  added in Presto 340.
+
+<!--more-->
+
+# What are original files?
+
+Files present in flat ORC tables have the standard ORC schema. When a flat table is converted into a transactional table, existing files are not converted into Hive ACID format.
+Such files, in a transactional table which are not in Hive ACID format, are called original files. These files are named as `000000_X`, `000000_X_copy_Y`. These files don’t have ACID
+columns and have differences in the schema as follows:
+
+|Table Schema       |Original File Schema|Delta File Schema             |
+|:------------------|:-------------------|:-----------------------------|
+|                   |                    |                              |
+|n_nationkey : int, |struct {            |struct {                      |
+|n_name : string,   |n_nationkey : int,  |operation : int,              |
+|n_regionkey : int, |n_name : string,    |originalTransaction : bigint, |
+|n_comment : string` |n_regionkey : int, |bucket : int,                 |
+|                   |n_comment : string  |rowId : bigint,               |
+|                   |}                   |currentTransaction : bigint,  |
+|                   |                    |row : struct {                |
+|                   |                    |    n_nationkey : int,        |
+|                   |                    |    n_name : string,          |
+|                   |                    |    n_regionkey : int,        |
+|                   |                    |    n_comment : string        |
+|                   |                    |  }                           |
+|                   |                    |}                             |
+|                   |                    |                              |
+
+
+
+Before Presto 340, Presto used to fail the query if it reads from a  Hive transactional table having original files.
+
+# Update and delete support on original files
+
+Hive achieves updates/deletes on a row in original files by synthetically generating ACID columns for original files. Presto follows the same mechanism of generating ACID columns
+synthetically as discussed later.
+
+## ACID Columns’ generation on original files
+
+Files in Hive ACID format have 5 ACID columns, but we need only 3 columns i.e. `originalTransactionId`, `bucketId` and `rowId` to uniquely identify a row. In this section we will
+see how these 3 columns are synthetically generated for  original files.
+
+### Original transaction ID
+
+An original Transaction ID is the write ID when a record is first created. For original files, the original transaction ID is always the starting value i.e. 0.
+
+### Bucket ID
+
+Bucket ID is retrieved from the original file name. For the original file `0000ABC_DEF` or `0000ABC_DEF_copy_G`, the bucket ID will be `DEF`.
```

Comment:
Not ABC ?

Reply:
yeah, thank you for pointing out. changed.